spiketoolkit/sortingcomponents/detection.py

```diff
@@ -191,62 +188,68 @@ def _detect_and_align_peaks_chunk(ii, rec_arg, chunks, channel_ids, detect_thres
-                    if idx_spike - n_pad > 0 and idx_spike + n_pad < len(trace):
-                        spike = trace[idx_spike - n_pad:idx_spike + n_pad]
-                        t_spike = np.arange(idx_spike - n_pad, idx_spike + n_pad)
-                    elif idx_spike - n_pad < 0:
-                        spike = trace[:idx_spike + n_pad]
-                        spike = np.pad(spike, (np.abs(idx_spike - n_pad), 0), 'constant')
-                        t_spike = np.arange(idx_spike + n_pad)
-                        t_spike = np.pad(t_spike, (np.abs(idx_spike - n_pad), 0), 'constant')
-                    elif idx_spike + n_pad > len(trace):
-                        spike = trace[idx_spike - n_pad:]
-                        spike = np.pad(spike, (0, idx_spike + n_pad - len(trace)), 'constant')
-                        t_spike = np.arange(idx_spike - n_pad, len(trace))
-                        t_spike = np.pad(t_spike, (0, idx_spike + n_pad - len(trace)), 'constant')
-
-                    if upsample > 1:
-                        spike_up = ss.resample(spike, int(upsample * len(spike)))
-                        t_spike_up = np.linspace(t_spike[0], t_spike[-1], num=len(spike_up))
-                    else:
-                        spike_up = spike
-                        t_spike_up = t_spike
-                    if detect_sign == -1:
-                        peak_idx = np.argmin(spike_up)
-                        peak_val = np.min(spike_up)
-                    elif detect_sign == 1:
-                        peak_idx = np.argmax(spike_up)
-                        peak_val = np.max(spike_up)
-                    else:
-                        peak_idx = np.argmax(np.abs(spike_up))
-                        peak_val = np.max(np.abs(spike_up))
-
-                    min_time_up = t_spike_up[peak_idx]
-                    sp_times[i].append(int(min_time_up))
-                    sp_amplitudes[i].append(peak_val)
+    # get thresholds
+    if detect_sign == -1:
+        thresholds = -detect_threshold * np.median(np.abs(traces) / 0.6745, 1)
+        crossings = traces < thresholds[:, None]
+    elif detect_sign == 1:
+        thresholds = detect_threshold * np.median(np.abs(traces) / 0.6745, 1)
+        crossings = traces > thresholds[:, None]
+    else:
+        thresholds = detect_threshold * np.median(np.abs(traces) / 0.6745, 1)
+        crossings = np.abs(traces) > thresholds[:, None]
+
+    spike_mask = np.diff(crossings.astype(np.int8), axis=1) > 0
+    spikes = np.asarray(np.where(spike_mask)).T
+
+    if not align:
+        for i, ch in enumerate(channel_ids):
+            spikes_ch_i = np.where(spikes[:, 0] == i)[0]
+            sp_times[i] = spikes[spikes_ch_i, 1]
+            sp_amplitudes[i] = traces[i, spikes[spikes_ch_i, 1]]
+    else:
+        for i, ch in enumerate(channel_ids):
+            spikes_ch_i = np.where(spikes[:, 0] == i)
+            spike_idx = spikes[spikes_ch_i[0], 1]
+
+            for idx_spike in spike_idx:
```

Comment:
horrible notation : swap pre and post fix to make scalar!!!!

Reply:
I dont understand, make code suggestion